api/v1alpha3/azureclusteridentity_conversion.go

```diff
@@ -30,25 +30,25 @@ const (
 )
 
 // ConvertTo converts this AzureCluster to the Hub version (v1beta1).
-func (src *AzureClusterIdentity) ConvertTo(dstRaw conversion.Hub) error {
+func (c *AzureClusterIdentity) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*infrav1beta1.AzureClusterIdentity)
-	if err := Convert_v1alpha3_AzureClusterIdentity_To_v1beta1_AzureClusterIdentity(src, dst, nil); err != nil {
+	if err := Convert_v1alpha3_AzureClusterIdentity_To_v1beta1_AzureClusterIdentity(c, dst, nil); err != nil {
```

Comment:
these files receiver names should not be changed and ignored.

Reply:
We can add following in the golangci lint yaml 
```
    - linters:
        - stylecheck
      text: "ST1016: methods on the same type should have the same receiver name"
      path: .*(api|types)\/.*\/*_types\.go$
```

Reply:
Add above after this line : 
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/.golangci.yml#L102